tools/fast_pusher_.py

```diff
@@ -98,28 +102,33 @@ def main():
   if not args.config and not args.tarball:
     raise Exception('Either --config or --tarball must be specified.')
 
+  if len(args.digest or []) != len(args.layer or []):
+    raise Exception('--digest and --layer must have matching lengths.')
+
   # If config is specified, use that.  Otherwise, fallback on reading
   # the config from the tarball.
   config = args.config
   if args.config:
+    logging.info('Reading config from %r', args.config)
     with open(args.config, 'r') as reader:
       config = reader.read()
   elif args.tarball:
+    logging.info('Reading config from tarball %r', args.tarball)
     with v2_2_image.FromTarball(args.tarball) as base:
       config = base.config_file()
 
-  if len(args.digest or []) != len(args.layer or []):
-    raise Exception('--digest and --layer must have matching lengths.')
-
-  transport = transport_pool.Http(httplib2.Http, size=_THREADS)
-
   # Resolve the appropriate credential to use based on the standard Docker
   # client logic.
+  logging.info('Loading Docker credentials for repository %r', str(name))
   creds = docker_creds.DefaultKeychain.Resolve(name)
 
+  transport = transport_pool.Http(httplib2.Http, size=_THREADS)
+
   with docker_session.Push(name, creds, transport, threads=_THREADS) as session:
+    logging.info('Loading v2.2 image from disk ...')
```

Comment:
I'd expect this tool to be on the most folks' critical path, but we should probably instrument more of the stuff in `tools`.